### PR TITLE
Added "find" option that allows for regex searching in recipebook

### PR DESCRIPTION
**Pull request by dreamsmasher**
Implements #36. The hard part wasn't actually implementing the search function, it was integrating it into the app ;)

Right now it only supports a single search string at a time, and doesn't output colour. Otherwise, it matches the specification laid out in #36. 

I had to add toStr to src/RichText's exports in order to get this to work, it's actually a useful function.

Builds successfully, passes all tests and elementary regex testing. Need to add unit tests too.

### CODE REVIEW

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:201]: This line is very long :smile: Let's go ahead and track colorization on https://github.com/LuxMiranda/herms/issues/36 or a follow-up, where we can fit more details, use labels, etc.

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:203]: Nit: You could inline `matches`, it's only used once and it's not a terribly complicated expression. 

(This is just a small/stylistic comment, feel free to address or not as you prefer, it won't hold up merging.)

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:387]: Looks like two spaces too many:
```suggestion
runWithOpts (Find regexp)                           = findRecipes regexp
```

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:418]: I feel like `regx` is an uncommon abbreviation - perhaps `regex` would be more obvious? Also, this is missing the closing `@`. 
```suggestion
-- | @findRegexP@ returns a parser for a regex string
```
(If you accept this suggestion, please also change the function name just below)

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:522]: Did you intentionally change the spacing here? If so, why?

File: app/herms.hs
Comment (langston-barrett) [inline app/herms.hs:525]: Did you intentionally change the spacing here? If so, why?

File: src/Find.hs
Comment (langston-barrett) [inline src/Find.hs:27]: There's a few extra lines at the end of this file - please remove them.

File: src/Find.hs
Comment (langston-barrett) [inline src/Find.hs:13]: I don't normally see the `in` of a `let` expression indented one more space - Could you re-indent it at the same level as the `let`?

File: src/Find.hs
Comment (langston-barrett) [inline src/Find.hs:3]: It would be great to have Haddock comments for the exported functions in this file, but it won't hold up merging.

File: src/Find.hs
Comment (langston-barrett) [inline src/Find.hs:13]: You could simplify this line (note - I haven't actually tested this, but I _think_ it's equivalent):
```suggestion
            in map (\b -> name ++ ": " ++ b) (findInRecipe rgx r)
```

File: src/Types.hs
Comment (langston-barrett) [inline src/Types.hs:322]: Why did this code change? Was it intentional? Does it have to do with the rest of this PR?

**Review by langston-barrett — CHANGES_REQUESTED** [review]
A few nits so far, but looks good overall. Looking forward to seeing the final state! Thanks @dreamsmasher!

File: app/herms.hs
Comment (dreamsmasher) [reply app/herms.hs]: Sorry about that! Autowrapping is a major hindrance sometimes

File: app/herms.hs
Comment (dreamsmasher) [reply app/herms.hs:523]: Some of the other lines were misaligned, just wanted to maintain consistency.

File: src/Find.hs
Comment (dreamsmasher) [reply src/Find.hs]: Sure thing, I usually indent `in` one space so that the following statement lines up vertically with the `let` blocks. Fixed.

Comment (dreamsmasher) [reply src/Find.hs]: Fixed, thank you for the suggestion!

**Review by dreamsmasher — COMMENTED** [review]
Changes addressed in 1fdcd1edfebc5d1f650c2c9a13566712bd272ded.

**Comment by dreamsmasher** [comment]
Added requested changes, and readded the Data.Monoid imports. Sorry about the failing build on GHC-8.2.2!

File: app/herms.hs
Comment (dreamsmasher) [reply app/herms.hs:204]: I think I'll keep this line as is - the control flow is more visible this way I think.

Comment (langston-barrett) [reply app/herms.hs:204]: Control flow is a tricky concept in Haskell :grin:

File: src/Types.hs
Comment (langston-barrett) [reply src/Types.hs:322]: Ah, I see. Can we keep the formatting closer to what it previously was so that the change is clearer from the diff?

**Comment by langston-barrett** [comment]
> Sorry about the failing build on GHC-8.2.2!

I think we can drop GHC 8.2.2, compatibility with three major versions is already pretty good. Feel free to make an issue and/or a pre-PR for that if you prefer it to fixing the 8.2.2 build.

**Comment by langston-barrett** [comment]
Thanks for addressing those comments, this is looking pretty good! I'll test locally and merge once the build passes and the final comment gets resolved.

File: src/Types.hs
Comment (dreamsmasher) [reply src/Types.hs:322]: Sure thing, will fix now.

**Comment by dreamsmasher** [comment]
Awesome, thank you so much @langston-barrett! All comments now resolved.

**Comment by langston-barrett** [comment]
@dreamsmasher It looks like the build is still failing. Could you remove GHC 8.2.2 from the build configuration?

**Comment by langston-barrett** [comment]
(We should also remove it from the supported list in the Cabal file)

**Comment by dreamsmasher** [comment]
All checks passed!

**Comment by dreamsmasher** [comment]
Also - any chance we could add the ‘hacktoberfest‘ tag back to this repo’s topics? New regulations just came out that only PR’s to repos with the tag will be accepted for the free shirts :/

**Comment by langston-barrett** [comment]
> Also - any chance we could add the ‘hacktoberfest‘ tag back to this repo’s topics? New regulations just came out that only PR’s to repos with the tag will be accepted for the free shirts :/

I made #124 to track this. Does it need to be done before merging this?

> All checks passed!

Thanks! I just need to try it out locally now.

File: src/Types.hs
Comment (langston-barrett) [inline src/Types.hs:327]: Let's go ahead and just delete this line

File: src/Types.hs
Comment (langston-barrett) [inline src/Types.hs:326]: Unnecessary parens:
```suggestion
        where serv = case maybeServings of 
```

File: src/Types.hs
Comment (langston-barrett) [inline src/Types.hs:324]: Instead of the `where` with the `case`, you can equivalently use 
```suggestion
    showRecipeHeader t r (maybeServings <|> Just 1)
```
where `<|>` comes from `Control.Applicative`.

**Review by langston-barrett — COMMENTED** [review]
A few more tiny changes, otherwise looks good.

**Comment by langston-barrett** [comment]
Thanks @dreamsmasher!

**Pull request merged**
